packages/core/src/routes/admin-user-social.ts

```diff
@@ -0,0 +1,108 @@
+import { notImplemented } from '@logto/cli/lib/connector/consts.js';
+import { ConnectorType, userInfoSelectFields, userProfileResponseGuard } from '@logto/schemas';
+import { has, pick } from '@silverhand/essentials';
+import { object, record, string, unknown } from 'zod';
+
+import RequestError from '#src/errors/RequestError/index.js';
+import koaGuard from '#src/middleware/koa-guard.js';
+import assertThat from '#src/utils/assert-that.js';
+
+import type { AuthedRouter, RouterInitArgs } from './types.js';
+
+export default function adminUserSocialRoutes<T extends AuthedRouter>(
+  ...[router, tenant]: RouterInitArgs<T>
+) {
+  const {
+    queries: {
+      users: { findUserById, updateUserById, hasUserWithIdentity, deleteUserIdentity },
+    },
+    connectors: { getLogtoConnectorById },
+  } = tenant;
+
+  router.post(
+    '/users/:userId/identities',
+    koaGuard({
+      params: object({ userId: string() }),
+      body: object({
+        connectorId: string(),
+        connectorData: record(string(), unknown()),
+      }),
+      status: [204, 404, 422],
+    }),
```

Comment:
should we also return the user info after updating its `identities`?

Reply:
I'm not sure. Initially I thought it was unnecessary to do so, but now I think maybe you are right...

Reply:
Updated